### DISCUSSION TITLE
Order By Relation - One to Many

### DISCUSSION THREAD

**Discussion by paulosalvatore** [prisma/prisma#6098] [Q&A]
## Models

Consider the following models:

```prisma
model EventGroup {
    id Int @id @default(autoincrement())

    events Event[]
}

model Event {
    id Int @id @default(autoincrement())

    eventGroup   EventGroup @relation(fields: [eventGroupId], references: [id])
    eventGroupId Int

    startDateTime DateTime
    endDateTime   DateTime
}
```

## What's working

I'm using `findMany` to get all `EventGroups` and including all their `Events`, ordered by `startDateTime` `asc`.

```ts
this.prisma.eventGroup.findMany({
    include: {
        events: {
            orderBy: {
                startDateTime: 'asc',
            },
        },
    },
});
```

## Problem

The problem is that I want to also order `EventGroup` by `Event.startDateTime`. Is that possible?
Something like:

```ts
this.prisma.eventGroup.findMany({
    include: {
        events: {
            orderBy: {
                startDateTime: 'asc',
            },
        },
    },
    orderBy: [
        {
            events: {
                startDateTime: 'asc',
            },
        },
    ],
});
```

## Error

But given that `events` isn't available at `EventGroupOrderByInput`, that isn't possible:

![image](https://user-images.githubusercontent.com/7906171/111064155-3dfc4780-8491-11eb-96b9-0f0103878d20.png)

## Preview-Feature: `orderByRelation`

Also, I have `orderByRelation` activated:

```prisma
generator client {
    provider = "prisma-client-js"
    previewFeatures = ["orderByRelation"]
}
```

## Question

I know that I can use `orderByRelation` to order by `Many to One` relations, but how can I do that with `One to Many`?

**Comment by ryands17** [comment]
I'm not sure if this is is possible in a one-to-many relationship as how would you order `EventGroup` by multiple `Event` start times? Can this be done via a raw query as well?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
